viam-cartographer_test.go

```diff
@@ -186,11 +223,183 @@ func TestNew(t *testing.T) {
+	})
+
+	t.Run("Failed creation of cartographer slam service with invalid sensor "+
+		"that errors during call to NextPointCloud when feature flag enabled", func(t *testing.T) {
+		termFunc := initTestCL(t, logger)
+		defer termFunc()
+
+		dataDirectory, fsCleanupFunc := initInternalState(t)
+		defer fsCleanupFunc()
+
+		attrCfg := &vcConfig.Config{
+			ModularizationV2Enabled: &_true,
+			Sensors:                 []string{"invalid_sensor"},
+			ConfigParams:            map[string]string{"mode": "2d"},
+			DataDirectory:           dataDirectory,
+			DataRateMsec:            testDataRateMsec,
+			UseLiveData:             &_true,
+		}
+
+		_, err = internaltesthelper.CreateSLAMService(t, attrCfg, logger, false, testExecutableName)
+		test.That(t, err, test.ShouldBeError,
+			errors.New("failed to get data from lidar: ValidateGetData timeout: invalid sensor"))
+	})
+
+	t.Run("Successful creation of cartographer slam service in localization mode when feature flag enabled", func(t *testing.T) {
+		termFunc := initTestCL(t, logger)
+		defer termFunc()
+
+		dataDirectory, fsCleanupFunc := initInternalState(t)
+		defer fsCleanupFunc()
+
+		attrCfg := &vcConfig.Config{
+			ModularizationV2Enabled: &_true,
+			Sensors:                 []string{"replay_sensor"},
+			ConfigParams:            map[string]string{"mode": "2d"},
+			DataDirectory:           dataDirectory,
+			UseLiveData:             &_true,
+			MapRateSec:              &_zeroInt,
+		}
+
+		svc, err := internaltesthelper.CreateSLAMService(t, attrCfg, logger, false, testExecutableName)
+		test.That(t, err, test.ShouldBeNil)
+
+		// TODO: Implement these
+		// timestamp1, err := svc.GetLatestMapInfo(context.Background())
+		// test.That(t, err, test.ShouldBeNil)
+		// _, err = svc.GetPointCloudMap(context.Background())
+		// test.That(t, err, test.ShouldBeNil)
+		// timestamp2, err := svc.GetLatestMapInfo(context.Background())
+		// test.That(t, err, test.ShouldBeNil)
+		// test.That(t, timestamp1.After(_zeroTime), test.ShouldBeTrue)
+		// test.That(t, timestamp1, test.ShouldResemble, timestamp2)
+
+		test.That(t, svc.Close(context.Background()), test.ShouldBeNil)
+	})
+
+	t.Run("Successful creation of cartographer slam service in non localization mode when feature flag enabled", func(t *testing.T) {
+		termFunc := initTestCL(t, logger)
+		defer termFunc()
+
+		dataDirectory, err := os.MkdirTemp("", "*")
+		test.That(t, err, test.ShouldBeNil)
+
+		attrCfg := &vcConfig.Config{
+			ModularizationV2Enabled: &_true,
+			Sensors:                 []string{"replay_sensor"},
+			ConfigParams:            map[string]string{"mode": "2d"},
+			DataDirectory:           dataDirectory,
+			UseLiveData:             &_false,
+			MapRateSec:              &testMapRateSec,
+		}
+
+		svc, err := internaltesthelper.CreateSLAMService(t, attrCfg, logger, false, testExecutableName)
+		test.That(t, err, test.ShouldBeNil)
+
+		// TODO: Implement these
+		// timestamp1, err := svc.GetLatestMapInfo(context.Background())
```

Comment:
These can't be implemented until the corresponding methods are supported when the feature flag is on.